### PR TITLE
Add support for .yml in addition to .yaml

### LINKED ISSUE #6668: quarkus-config-yaml does not recognize "application.yml"
**Describe the bug**
While the plugin recognizes `application.yaml`, it does not recognize `application.yml`, and thus does not get the configuration as normally expected. (`.yml` is accepted by Spring, for instance)

**Expected behavior**
Quarkus would pick up the configuration placed in `application.yml`

**Actual behavior**
Quarkus does not pick up the configuration placed in `application.yml`

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #6668

### CODE REVIEW

**Comment by geoand** [comment]
@dmlloyd was there any specific reason that `.yml` wasn't added initially?

File: docs/src/main/asciidoc/config.adoc [hunk 1]
```diff
@@ -625,6 +625,11 @@ quarkus:
     password: quarkus
 ----
 
+[NOTE]
+====
+Quarkus also supports using `application.yml` as the name of YAML file. The same rules apply for this file as for `application.yaml`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/config.adoc]:
```suggestion
Quarkus also supports using `application.yml` as the name of the YAML file. The same rules apply for this file as for `application.yaml`.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks OK to me except for a minor gripe on the doc.

File: docs/src/main/asciidoc/config.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/config.adoc]: Good one! A classic mistake I always make :). You can probably spot those with closed eyes now :P

**Review by gsmet — APPROVED** [review]

**Comment by dmlloyd** [comment]
Just avoiding confusion I guess.  What about microprofile-config.yml?

**Comment by geoand** [comment]
Should we add `microprofile-config.yml` or wait until someone complains?

**Comment by gsmet** [comment]
I would say let's leave it at that.

**Comment by geoand** [comment]
> I would say let's leave it at that.

@dmlloyd OK with this approach?

**Review by dmlloyd — APPROVED** [review]

**Pull request merged**